dbms/src/Core/Settings.h

```diff
@@ -345,6 +345,9 @@ struct Settings : public SettingsCollection<Settings>
     M(SettingBool, check_query_single_value_result, true, "Return check query result as single 1/0 value") \
     M(SettingBool, allow_drop_detached, false, "Allow ALTER TABLE ... DROP DETACHED PART[ITION] ... queries") \
     \
+    M(SettingSeconds, replica_error_decrease_period, DBMS_CONNECTION_POOL_WITH_FAILOVER_DEFAULT_DECREASE_ERROR_PERIOD, "Time period reduces replica error counter by 2 times.") \
+    M(SettingUInt64, replica_error_max_count, DBMS_CONNECTION_POOL_WITH_FAILOVER_MAX_ERROR_COUNT, "Max number of errors per replica, prevents piling up increadible amount of errors if replica was offline for some time and allows it to be reconsidered in a shorter amount of time.") \
```

Comment:
We probably should prefix the names with something like 'distributed' to emphasize that these parameters control the Distributed table engine, and to avoid confusion with ReplicatedMergeTree replicas, Dictionary replicas and maybe some other replicas we might have.

About the names themselves, I'd name the first one 'distributed_replica_error_half_life'. This name makes it immediately clear that the error count decays exponentially, and the parameter controls the time after which the error count is reduced twice.
For the second one, I'd use 'distributed_replica_error_cap', to convey that the error count is capped at this value. 'max_count' is less precise, it may mean for example that the replica is disconnected after this number is reached, or something else.

Reply:
Agree, fixed